src/generic/gridsel.cpp

```diff
@@ -48,13 +48,55 @@ wxGridSelection::wxGridSelection( wxGrid * grid,
 {
     m_grid = grid;
     m_selectionMode = sel;
+    m_isSelecting = false;
 }
 
 bool wxGridSelection::IsSelection()
 {
     return !m_selection.empty();
 }
 
+
+void wxGridSelection::StartSelecting()
+{
+    m_isSelecting = true;
+}
+void wxGridSelection::EndSelecting()
+{
+    m_isSelecting = false;
+    //send the RANGE_SELECTED events
+    if (IsSelection())
+    {
+        size_t n;
+        wxRect r;
+        wxGridCellCoords coords1, coords2;
+
+        // deselect all blocks and update the screen
+        while ( ( n = m_selection.size() ) > 0)
+        {
+            n--;
+            const wxGridBlockCoords& block = m_selection[n];
+            coords1 = block.GetTopLeft();
+            coords2 = block.GetBottomRight();
+            if ( !m_grid->GetBatchCount() )
+            {
+                wxGridRangeSelectEvent gridEvt( m_grid->GetId(),
+                                                wxEVT_GRID_RANGE_SELECTED,
+                                                m_grid,
+                                                coords1,
+                                                coords2,
+                                                true );
+                m_grid->GetEventHandler()->ProcessEvent(gridEvt);
+            }
+        }
+    }
+    else
+    {
+        ClearSelection();
+    }
```

Comment:
Is this really necessary? We have no selection anyhow, so why clear it?

Reply:
I believe it's needed to send the final SELECTED event with the empty selection.   Not 100% sure though.